### PR TITLE
Adding some metadata to the task field for reporting upstream.

### PR DESCRIPTION
**Pull request by kblibr**

### CODE REVIEW

**Comment by addisonj** [comment]
This will need a rebase.

Also, I think for the build to pass, we will need to get simple-swf published first

File: src/workers/ActivityWorker.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class ActivityWorker extends SWFActivityWorker implements LogWorkerMixin
     this.logInfo('task completed', this.buildTaskMeta(task, { details: details }))
   }
   onTaskFailed(task: ActivityTask, execution: Activity, err: Error, details: TaskStatus) {
-    const taskInfo = {type: task.activityName(), id: execution.id}
+    const taskInfo = {type: task.activityName(), id: execution.id, control: task.getControl()}
```
Comment (addisonj) [inline src/workers/ActivityWorker.ts]: As mentioned in the review for simple-swf can we also add this to onWorkflowFailed so they have the same info?

Comment (kblibr) [reply src/workers/ActivityWorker.ts]: I've added it to all the places where TaskInfo exists.

Comment (addisonj) [reply src/workers/ActivityWorker.ts]: We also want this on `src/workers/DeciderWorker.ts:49` where we call `workflowFailed` to also get the context data

Comment (kblibr) [reply src/workers/ActivityWorker.ts]: sorry.  I looked all over the ActivityWorker for workflow failed :)

File: package.json [hunk 2]
```diff
@@ -67,7 +67,7 @@
     "lodash": "^4.10.0",
     "raven": "^2.0.2",
     "shortid": "^2.2.6",
-    "simple-swf": "^0.3.2",
+    "simple-swf": "^0.4.0",
```
Comment (addisonj) [inline package.json]: we should probably pin this to 0.4.0 and take explicit updates, now that I think about it

**Pull request merged**